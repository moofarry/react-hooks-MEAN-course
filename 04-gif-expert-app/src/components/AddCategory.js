import React from "react";
import PropTypes from "prop-types";

export const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = React.useState('');

  const handleInputChange = (e) => {
    setInputValue(e.target.value);
  };

  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((prevCategories) => [ inputValue,...prevCategories]);
      setInputValue("");
    }
  };

  return (
    <form onSubmit={handleSubmit}>
      <input
        value={inputValue}
        type="text"
        onChange={handleInputChange}
      ></input>
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};
