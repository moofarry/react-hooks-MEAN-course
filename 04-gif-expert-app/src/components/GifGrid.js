import React from "react";

export const GifGrid = ({ category }) => {
  const [count, setCount] = React.useState(0);

  React.useEffect(() => {
    getGifs();
  });

  const getGifs = async () => {
    const url =
      "https://api.giphy.com/v1/gifs/search?q=Uribe&limit=10&api_key=Enhu5gM26OHBLH0hYgf1tKNVcT2JHheo";
    const resp = await fetch(url);
    const { data } = await resp.json();

    const gifs = data.map(({ id, title, images }) => {
      return {
        id: id,
        title: title,
        url: images?.downsized_medium.url,
      };
    });

    console.log(`gifs`, gifs);
  };
  return (
    <div>
      <h3>{category}</h3>

      <h3>{count}</h3>
      <button
        onClick={() => {
          setCount(count + 1)}}
      ></button>
    </div>
  );
};
