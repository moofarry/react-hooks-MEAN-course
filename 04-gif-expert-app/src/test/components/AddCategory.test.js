import React from "react";
import "@testing-library/jest-dom";
import { shallow } from "enzyme";
import { AddCategory } from "../../components/AddCategory";

describe("test in <AddCategory/>", () => {
  const setCategories = jest.fn();
  let wrapper = shallow(<AddCategory setCategories={setCategories} />);
  const input = wrapper.find("input");
  const valueInput = "hello world";

  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });

  test("should show correcly", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("should change text box", () => {
    input.simulate("change", { target: { valueInput } });
  });

  test("Dont post the information on submit", () => {
    wrapper.find("form").simulate("submit", { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  // test("should call setCategory and clean text box", () => {
  //   wrapper.find("input").simulate("change", { target: { valueInput } });
  //   wrapper.find("form").simulate("submit", { preventDefault() {} });

  //   expect(setCategories).toHaveBeenCalled();
  //   expect(setCategories).toHaveBeenCalledTimes(1);
  //   expect(setCategories).toHaveBeenCalledWith(expect.any(Function));
  //   expect(wrapper.find("input").prop('value')).toBe("");
  // });

});
