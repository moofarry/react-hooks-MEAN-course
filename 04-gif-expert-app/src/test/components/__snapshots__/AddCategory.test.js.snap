// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test in <AddCategory/> should show correcly 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
</form>
`;
